Documentation/ceph-best-practices.md

```diff
@@ -0,0 +1,516 @@
+requests on a per-OSD basis. The node-level recommendation we have is given below where
+`osd_memory_target`. This is a Ceph configuration described more afterwards.
+
+> `Total RAM required = [number of OSDs] x (1 GB + osd_memory_target) + 16 GB `
+
+Ceph OSDs will attempt to keep heap memory usage under a designated target size set via the
+`osd_memory_target` configuration option. Ceph’s default `osd_memory_target` is 4GB, and it is
+recommended to never decrease the `osd_memory_target` below 4GB. You may wish to increase this value
+to improve overall Ceph read performance by allowing the OSDs to use more RAM. While the total
+amount of heap memory mapped by the process should generally stay close to this target, there is no
+guarantee that the Linux kernel will actually reclaim memory that has been unmapped.
+
+As an example, for a node hosting 8 OSDs, memory Requests would be calculated as below.
+
+> `8 OSDs x (1GB + 4GB) + 16GB = 56GB per node`
+
+This gives us resource usage for each OSD.
+
+> `56GB / 8 OSDs = 7GB`
+
+Ceph has a feature that allows it to set `osd_memory_target` automatically when a Rook OSD Resource
+Request is set. However, Ceph sets this value 1:1 and does not leave overhead for waiting for the
+kernel to free memory. Therefore, we recommend setting `osd_memory_target` in Ceph explicitly, even
+if you wish to use the default value. Set Rook’s OSD resource requests accordingly and to a higher
+value than `osd_memory_target` by at least an additional 1GB. This is so Kubernetes does not
+schedule more applications or Ceph daemons onto a node than the node is likely to have RAM available for.
+
+OSD RAM Resource Requests come with the same cluster-wide Resource Requests note as for OSD CPU. Use
+the highest Requests for a cluster consisting of multiple different configurations of OSDs.
+
+### Resource Requests - Gateways
+
+For gateways, the best recommendation is to always tune your workload and daemon configurations.
+However, we do recommend the following initial configurations:
+
+**RGWs:**
+ * 6-8 CPU cores
+ * 64 GB RAM (32 GB minimum – may only apply to older "civetweb" protocol)
+
+> **NOTE:** The numbers below for RGW assume a lot of clients connecting. Thus they might not be the
+> best for your scenario. Additionally, the RAM usage should be lower for the newer “beast” protocol
+> compared to the older “civetweb” protocol.
+
+**MDS:**
+ * 2.5 GHz CPU with a least 2 cores
+ * 3GB RAM
+
+**NFS-Ganesha:**
+ * 6-8 CPU cores (untested, high estimate)
+ * 4GB RAM for default settings (settings hardcoded in Rook presently)
+
+
+## Basic Performance Enhancements
+
+The following are some basic performance enhancements. These are a few easy, low-hanging-fruit
+recommendations.
+
+> **NOTE:** Not all of these will be right for all clusters or workloads. Always performance test
+> and use your best judgment.
+
+* You can gain performance by using a CNI plugin with an accelerated network stack. For example,
```

Comment:
Probably worth mentioning that Rook supports Multus too so that we can expose dedicated network interface without enabling host networking.